testsupport/wait/member.go

```diff
@@ -246,6 +246,117 @@ func (a *MemberAwaitility) WaitForUserAccount(t *testing.T, name string, criteri
+}
+
+// UntilSpaceRequestHasTierName returns a `SpaceRequestWaitCriterion` which checks that the given
+// SpaceRequest has a specific tierName
+func UntilSpaceRequestHasTierName(expected string) SpaceRequestWaitCriterion {
+	return SpaceRequestWaitCriterion{
+		Match: func(actual *toolchainv1alpha1.SpaceRequest) bool {
+			return actual.Spec.TierName == expected
+		},
+		Diff: func(actual *toolchainv1alpha1.SpaceRequest) string {
+			return fmt.Sprintf("expected tier name to be '%s'\nbut it was '%s'", expected, actual.Spec.TierName)
+		},
+	}
+}
+
+// UntilSpaceRequestHasTargetClusterRoles returns a `SpaceRequestWaitCriterion` which checks that the given
+// SpaceRequest has the expected cluster roles
+func UntilSpaceRequestHasTargetClusterRoles(expected []string) SpaceRequestWaitCriterion {
+	return SpaceRequestWaitCriterion{
+		Match: func(actual *toolchainv1alpha1.SpaceRequest) bool {
+			return reflect.DeepEqual(expected, actual.Spec.TargetClusterRoles)
+		},
+		Diff: func(actual *toolchainv1alpha1.SpaceRequest) string {
+			return fmt.Sprintf("expected space roles to match:\n%s", Diff(expected, actual.Spec.TargetClusterRoles))
+		},
+	}
+}
+
+// UntilSpaceRequestHasStatusTargetClusterURL returns a `SpaceRequestWaitCriterion` which checks that the given
+// SpaceRequest has the expected .Status.TargetClusterURL value
+func UntilSpaceRequestHasStatusTargetClusterURL(expected string) SpaceRequestWaitCriterion {
+	return SpaceRequestWaitCriterion{
+		Match: func(actual *toolchainv1alpha1.SpaceRequest) bool {
+			return expected == actual.Status.TargetClusterURL
+		},
+		Diff: func(actual *toolchainv1alpha1.SpaceRequest) string {
+			return fmt.Sprintf("expected space roles to match:\n%s", Diff(expected, actual.Status.TargetClusterURL))
+		},
+	}
+}
+
+// UntilSpaceRequestHasConditions returns a `SpaceRequestWaitCriterion` which checks that the given
+// SpaceRequest has exactly all the given status conditions
+func UntilSpaceRequestHasConditions(expected ...toolchainv1alpha1.Condition) SpaceRequestWaitCriterion {
+	return SpaceRequestWaitCriterion{
+		Match: func(actual *toolchainv1alpha1.SpaceRequest) bool {
+			return test.ConditionsMatch(actual.Status.Conditions, expected...)
+		},
+		Diff: func(actual *toolchainv1alpha1.SpaceRequest) string {
+			return fmt.Sprintf("expected conditions to match:\n%s", Diff(expected, actual.Status.Conditions))
+		},
+	}
+}
+
+func matchSpaceRequestWaitCriterion(actual *toolchainv1alpha1.SpaceRequest, criteria ...SpaceRequestWaitCriterion) bool {
+	for _, c := range criteria {
+		if !c.Match(actual) {
+			return false
+		}
+	}
+	return true
+}
+
+func (a *MemberAwaitility) printSpaceRequestWaitCriterionDiffs(t *testing.T, actual *toolchainv1alpha1.SpaceRequest, criteria ...SpaceRequestWaitCriterion) {
+	buf := &strings.Builder{}
+	if actual == nil {
+		buf.WriteString("failed to find SpaceRequest\n")
+		buf.WriteString(a.listAndReturnContent("SpaceRequest", a.Namespace, &toolchainv1alpha1.UserAccountList{}))
```

Comment:
```suggestion
		buf.WriteString(a.listAndReturnContent("SpaceRequest", "", &toolchainv1alpha1.SpaceRequestList{}))
```

Reply:
Thanks! applied in [32f2a7c](https://github.com/codeready-toolchain/toolchain-e2e/pull/685/commits/32f2a7cc9eb5347d00c19dbbd3a44a54ecbcfb4c)